### ISSUE TITLE
Can't connect to SQL Database Utility: Error building component SQLDatabase: can't load plugin: sqlalchemy.dialects:postgres

### ISSUE THREAD

**Issue by antoremin** [langflow-ai/langflow#2438]
**Describe the bug**
I'm putting in a postgresql connection string and get an error: Error building component SQLDatabase: can't load plugin: sqlalchemy.dialects:postgres. 

**Browser and Version**
 - Browser Chrome
 - Version 1.05

**To Reproduce**
1. Make a new flow
2. Click on SQLDatabase utility
3. Put in postgresql URI connection string and click "run" 

**Screenshots**
<img width="420" alt="Screenshot 2024-06-28 at 3 07 30 PM" src="https://github.com/langflow-ai/langflow/assets/6918736/543d079b-2f22-4ccc-ae5f-c296bfc082a2">

**Additional context**
If I'm not mistaken, sqlalchemy deprecated postgres and only allows postgresql dialect. Yet in[ Langflow SQLDatabase](https://github.com/langflow-ai/langflow/blob/bcf5807262fd20e83d17b91c1d066a24a2fda4b4/src/backend/base/langflow/components/langchain_utilities/SQLDatabase.py#L17), SQLExecutor, etc, the postgresql URI is explicitly replaced with postgres: 
```
    def clean_up_uri(self, uri: str) -> str:
        if uri.startswith("postgresql://"):
            uri = uri.replace("postgresql://", "postgres://")
        return uri.strip()
```

**Comment by carlosrcoelho** [comment]
Hi @antoremin  ,

Do you need any assistance with this case? If not, please let us know if this issue can be closed.

**Comment by Alperemrehas** [comment]
> Hi @antoremin ,
> 
> Do you need any assistance with this case? If not, please let us know if this issue can be closed.

Actually I was having the same error yesterday.

**Comment by carlosrcoelho** [comment]
@antoremin 

I checked it out and the component seems outdated. It needs to be updated to the new model that the components are built on.

**Comment by Alperemrehas** [comment]
This solved my problem: from langchain_experimental.sql.base import SQLDatabase

from langflow.interface.custom.custom_component import CustomComponent


class SQLDatabaseComponent(CustomComponent):
    display_name = "SQLDatabase"
    description = "SQL Database"

    def build_config(self):
        return {
            "uri": {"display_name": "URI", "info": "URI to the database."},
        }
#SQLAlchemy 1.4 removed the deprecated postgres dialect name, the name postgresql must be used instead now.
    def clean_up_uri(self, uri: str) -> str:
        if uri.startswith("postgres://"):
            uri = uri.replace("postgres://", "postgresql://")
        return uri.strip()

    def build(self, uri: str) -> SQLDatabase:
        uri = self.clean_up_uri(uri)
        return SQLDatabase.from_uri(uri)

**Comment by Alperemrehas** [comment]
> @Alperemrehas Thank you!
> 
> Fixed here: #3292

Thank you @carlosrcoelho .

**Comment by bohdanleskovets** [comment]
Have the save problem while connecting to Redsfit database
My DB url: postgres://user:pass@host:5439/database

I tried many url options, but none worked. redshift://user, postgres+psycopg2://user and many other options that I found when trying to connect. None of them worked. Can someone tell me what the problem is and how to connect to the redshift DB?
